app.py

```diff
@@ -880,19 +881,9 @@ def tree_process(tree : ParentedTree, senttok: List[str]) -> tuple[ParentedTree,
 		# if initial parse labels non-gaps as GAP, change to N-Head by default
 		if subt.label.startswith('GAP') and senttok[i] != '_.':
 			subt.label = 'N-Head'
-		# if the label is a punctuation sequence, make sure it has a function ('p' by default)
-		# (handles cases where token is changed to punctuation in the text window)
-		if is_possible_punct_token(subt.label) and subt.label == '':
-			subt.label = SYMBOL_TAG+"-p"
-		# if the token is unambiguously punctuation, make sure it has 'p' function
-		if is_possible_punct_token(senttok[i]) and senttok[i] not in AMBIG_SYM:
-			subt.label = SYMBOL_TAG+"-p"
-		# if the token has function 'p' and has an idiosyncratic POS tag, change to that POS tag
-		if senttok[i] in np.concatenate(list(PUNCT_LABELS.values())) and is_punct_label(subt.label):
-			for label, tokens in PUNCT_LABELS.items():
-				if senttok[i] in tokens:
-					subt.label = label
-					break
+		# if label is a punctuation sequence or punctuation + "-p": assign the appropriate punctuation label (either from PUNCT_TAGS or the default SYMBOL_TAG)
+		if (is_possible_punct_token(subt.label) or is_punct_label(subt.label)) or (is_possible_punct_token(senttok[i]) and senttok[i] not in AMBIG_SYM):
```

Comment:
Trying to understand the 3 conditions. The first one is passing a label to a function that expects a token. Is it for cases where the token is not punct, but the label (tag) is?

Reply:
2nd condition: will this mean `'*-p'` is assigned if the token is NOT punctuation and the tag is `RRB-p`, for example?